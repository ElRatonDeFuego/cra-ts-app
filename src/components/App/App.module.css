.app-header {
  background-color: #282c34;
  font-size: calc(10px + 2vmin);
}

.app-link {
  color: #61dafb;
}
