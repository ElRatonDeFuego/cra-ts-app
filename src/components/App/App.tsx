import { faBarsStaggered } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styles from "./App.module.css";
import logo from "./logo.svg";

export const App = () => (
  <div className="text-center">
    <header className="bg-gray-900 min-h-screen flex flex-col items-center justify-center text-white text-3xl">
      <img
        src={logo}
        className={`h-96 pointer-events-none ${
          styles["app-logo"] ?? /* istanbul ignore next */ ""
        }`}
        alt="logo"
      />
      <p>
        <FontAwesomeIcon className="mr-6" icon={faBarsStaggered} />
        Edit <code>src/App.tsx</code> and save to reload.
      </p>
      <a
        className="text-blue-400"
        href="https://reactjs.org"
        target="_blank"
        rel="noopener noreferrer"
      >
        Learn React
      </a>
    </header>
  </div>
);
