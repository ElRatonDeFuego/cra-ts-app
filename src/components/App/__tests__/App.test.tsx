import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { App } from "../App";

test("renders learn react link", () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/iu);
  expect(linkElement).toBeInTheDocument();
});

test("should match the snapshot", () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});
