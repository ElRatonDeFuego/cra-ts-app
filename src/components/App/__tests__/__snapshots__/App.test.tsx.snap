// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="text-center"
>
  <header
    className="bg-gray-900 min-h-screen flex flex-col items-center justify-center text-white text-3xl"
  >
    <img
      alt="logo"
      className="h-96 pointer-events-none app-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      className="text-blue-400"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
